packages/legacy-credentials/src/vcInterop.ts

```diff
@@ -0,0 +1,180 @@
+/**
+ * Copyright (c) 2018-2023, BOTLabs GmbH.
+ *
+ * This source code is licensed under the BSD 4-Clause "Original" license
+ * found in the LICENSE file in the root directory of this source tree.
+ */
+
+import {
+  hexToU8a,
+  stringToU8a,
+  u8aCmp,
+  u8aToHex,
+  u8aToString,
+} from '@polkadot/util'
+import { base58Decode, base58Encode, blake2AsU8a } from '@polkadot/util-crypto'
+
+import { CType } from '@kiltprotocol/core'
+import { KiltCredentialV1, Types, constants } from '@kiltprotocol/vc-export'
+
+import type { ICType, IClaim, ICredential } from '@kiltprotocol/types'
+
+import { makeStatementsJsonLD } from './utils.js'
+
+/**
+ * Produces an instance of [[KiltAttestationProofV1]] from an [[ICredential]].
+ *
+ * @param credential Input credential.
+ * @param opts Additional parameters required for creating a proof from an [[ICredential]].
+ * @param opts.blockHash Hash of a block at which the proof must be verifiable.
+ * @returns An embedded proof for a verifiable credential derived from the input.
+ */
+export function proofFromICredential(
+  credential: ICredential,
+  { blockHash }: { blockHash: Uint8Array }
+): Types.KiltAttestationProofV1 {
+  // `block` field is base58 encoding of block hash
+  const block = base58Encode(blockHash)
+  // `commitments` (claimHashes) are base58 encoded in new format
+  const commitments = credential.claimHashes.map((i) =>
+    base58Encode(hexToU8a(i))
+  )
+  // salt/nonces must be sorted by statement digest (keys) and base58 encoded
+  const salt = Object.entries(credential.claimNonceMap)
+    .map(([hsh, slt]) => [hexToU8a(hsh), stringToU8a(slt)])
+    .sort((a, b) => u8aCmp(a[0], b[0]))
+    .map((i) => base58Encode(i[1]))
+  return {
+    type: constants.ATTESTATION_PROOF_V1_TYPE,
+    block,
+    commitments,
+    salt,
+  }
+}
+
+/**
+ * Transforms an [[ICredential]] object to conform to the KiltCredentialV1 data model.
+ *
+ * @param input An [[ICredential]] object.
+ * @param options Additional required and optional parameters for producing a VC from an [[ICredential]].
+ * @param options.issuer The issuer of the attestation to this credential (attester).
+ * @param options.timestamp Timestamp of the block referenced by blockHash in milliseconds since January 1, 1970, UTC (UNIX epoch).
+ * @param options.chainGenesisHash Optional: Genesis hash of the chain against which this credential is verifiable. Defaults to the spiritnet genesis hash.
+ * @returns A KiltCredentialV1 with embedded KiltAttestationProofV1 proof.
+ */
+export function vcFromICredential(
```

Comment:
I guess we can simply avoid exporting `*FromICredential`, right? I haven't seen those imported anywhere else, and anyway they are not exported from the index.

Reply:
that's true, I thought they may be imported in tests, but currently they are not

Reply:
https://github.com/KILTprotocol/sdk-js/pull/786/commits/7cb673598639334c520a65606b7973e4dac4caf4